src/search_iterator.cpp

```diff
@@ -97,10 +99,28 @@ search_iterator search_iterator::operator--(int) {
     return it;
 }
 
-std::string search_iterator::get_url() const {
+std::string search_iterator::get_path() const {
+    if ( ! internal ) {
+        return "";
+    }
+
+    std::string path = internal->get_document().get_data();
+    std::string dbDataType = internal->search->internal->database.get_metadata("data");
```

Comment:
If `data` is empty, we must default to "fullPath".

We should probably never have a `hasNewNamespaceScheme` zim without a "data" in the xapian db, but it is better to properly handle this anyway.

Reply:
@mgautierfr  Do we need to check this explicitly? Currently the namespace is trimmed(inside if block) only if `hasNewNamespace` is true and `dbDataType` is equal to "fullPath". In all other cases we will default to untrimmed data.
So if data is empty, we are anyway returning the untrimmed path

Reply:
If we are dealing with zim file created only with libzim.
I don't know any other writer, but it is not a big deal to check if data is empty and set it to "fullPath".

Reply:
aah yes, understood. Will do that :+1: